packages/material-react-table/src/components/inputs/MRT_FilterTextField.tsx

```diff
@@ -179,7 +179,7 @@ export const MRT_FilterTextField = <TData extends MRT_RowData>({
       textFieldProps.type === 'date'
         ? event.target.valueAsDate
         : textFieldProps.type === 'number'
-          ? event.target.valueAsNumber
+          ?  isNaN(event.target.valueAsNumber) ? undefined : event.target.value
```

Comment:
shouldn't this still return event.target.valueAsNumber when valid instead of just `value`?

Reply:
The columnFilters are not functioning as expected when the value is a **numeric 0**,
 while they work correctly when the value is the **text '0'**.
 
 please check the **CSB** and **Video**

Reply:
I am working in ERP application user will check  > 0 , >=0, <0,<=0.

Zero is not working when number input it is working as text input.